src/transformers/trainer.py

```diff
@@ -455,7 +455,15 @@ def train(self, model_path: Optional[str] = None):
             self.tb_writer.close()
 
         logger.info("\n\nTraining completed. Do not forget to share your model on huggingface.co/models =)\n\n")
-        return TrainOutput(global_step, tr_loss / global_step)
+        return TrainOutput(self.global_step, tr_loss / self.global_step)
+
+    def _log(self, logs: Dict[str, torch.Tensor]) -> None:
```

Comment:
We created a separate function as we now log in 2 places:
* training data in `train`
* evaluation data in `evaluate`

This is also where we add `epoch`.